godtools/App/Features/Menu/Presentation/Account/AccountViewModel.swift

```diff
@@ -8,87 +8,111 @@
-                self?.isLoadingProfile.accept(value: false)
-                self?.profileName.accept(value: AnimatableValue(value: profileNameDomainModel.value, animated: true))
+                
+                self?.isLoadingProfile = false
+                self?.profileName = profileNameDomainModel.value
+            }
+            .store(in: &cancellables)
+        
+        getGlobalActivityThisWeekUseCase.getGlobalActivityPublisher()
+            .receiveOnMain()
+            .sink { [weak self] (globalActivityThisWeekDomainModels: [GlobalActivityThisWeekDomainModel]) in
+                
+                self?.isLoadingGlobalActivityThisWeek = false
+                self?.globalActivityThisWeekDomainModels = globalActivityThisWeekDomainModels
+                self?.numberOfGlobalActivityThisWeekItems = globalActivityThisWeekDomainModels.count
             }
             .store(in: &cancellables)
     }
     
-    private func getAnalyticsScreenName (page: Int) -> String {
-        let accountItem: AccountItem = accountItems.value[page]
+    private func trackSectionViewedAnalytics(sectionName: String) {
         
-        return accountItem.analyticsScreenName
+        let trackScreen = TrackScreenModel(
+            screenName: sectionName,
+            siteSection: "account",
+            siteSubSection: "",
+            contentLanguage: getSettingsPrimaryLanguageUseCase.getPrimaryLanguage()?.analyticsContentLanguage,
+            secondaryContentLanguage: getSettingsParallelLanguageUseCase.getParallelLanguage()?.analyticsContentLanguage
+        )
+        
+        analytics.pageViewedAnalytics.trackPageView(trackScreen: trackScreen)
     }
+}
+
+// MARK: - Inputs
+
+extension AccountViewModel {
     
-    private var analyticsSiteSection: String {
-        return "account"
+    @objc func backTapped() {
+        flowDelegate?.navigate(step: .backTappedFromMyAccount)
     }
     
-    private var analyticsSiteSubSection: String {
-        return ""
+    func activityTapped() {
+        
+        // TODO: Ensure this is the correct page name for when Activity section is viewed. GT-1861 ~Levi
+        trackSectionViewedAnalytics(sectionName: "Activity")
     }
     
-    private func reloadAccountItems() {
+    func globalActivityTapped() {
         
-        let items: [AccountItem] = [
-            AccountItem(
-                itemId: .activity,
-                title: localizationServices.stringForMainBundle(key: "account.activity.title"),
-                analyticsScreenName: "Global Dashboard"
-            )
-        ]
-        
-        accountItems.accept(value: items)
+        trackSectionViewedAnalytics(sectionName: "Global Dashboard")
     }
     
-    func settingsTapped() {
-        print("TODO: AccountViewModel Implement settings.")
-    }
-
-    func accountPageDidAppear(page: Int) {
+    func globalActivityAnalyticsItemWillAppear(index: Int) -> AccountGlobalActivityAnalyticsItemViewModel {
```

Comment:
Naming things `willAppear` seems more event-driven/UIKit-y.  Since we're using SwiftUI now, it might be good to do away with this naming pattern and just call it `getGlobalActivityAnalyticsItem`.  Thoughts?

Reply:
Yeah you're right it does seem UIKit-y.  I will update this as well.  There will be some other places in the app that can be updated as we go along too.